project/src/main/java/com/google/sps/utilities/DatabaseWrapper.java

```diff
@@ -123,4 +128,45 @@ public void insertVolunteeringOpportunity(VolunteeringOpportunity opportunity) {
     mutations.add(builder.build());
     return mutations;
   }
+
+  /**
+   * Given an opportunityId, retrieve the corresponding volunteering opportunity
+   *
+   * @param opportunityId opportunityId of the opportunity to retrieve
+   * @return volunteering opportunity wrapped in a {@link Optional}
+   */
+  public Optional<VolunteeringOpportunity> getVolunteeringOpportunityForOppportunityId(
+      String opportunityId) {
+    SpannerOptions options = SpannerOptions.newBuilder().build();
+    Spanner spanner = options.getService();
+    DatabaseId db = DatabaseId.of(options.getProjectId(), instanceId, databaseId);
+    DatabaseClient dbClient = spanner.getDatabaseClient(db);
```

Comment:
Are you going to use injection for DatabaseWrapper? if so, these operations and spanner.close does not need to be here.

Reply:
Ack, I can make a new PR to do this from master since it would effect other functions as well. So if I understand correctly, I can refactor this by creating a class called DatabaseClient.java and passing in DatabaseClient in the constructor on the DatabaseWrapper class. And I can put all these operations in the constructor and put spanner.close() in a DatabaseClient function that can be called by DatabaseWrapper?

Reply:
Created a PR for this (no injection) on master and will make change here after approval.